spec/unit/puppet/provider/mysql_user/mysql_spec.rb

```diff
@@ -179,7 +179,9 @@
 
   describe 'destroy' do
     it 'removes a user if present' do
-      provider.class.expects(:mysql_caller).with("DROP USER 'joe'@'localhost'", 'system')
+      if_exists = mysqld_version >= '5.7' ? 'IF EXISTS' : ''
+
+      provider.class.expects(:mysql_caller).with("DROP USER #{if_exists} 'joe'@'localhost'", 'system')
```

Comment:
we'll still want to test the original query, so adding another test case would be most helpful here. there is also a pattern near the top of this file for setting the version number in the class.
e.g.
```ruby
it 'removes a user using IF EXISTS' do
  {set version to 5.7.1 or something}
  provider.class.expects(:mysql_caller).with("DROP USER IF EXISTS 'joe'@'localhost'", 'system')
   ...
end